bench_test.go

```diff
@@ -45,6 +45,16 @@ func benchmarkSet(c Geche[string, string], testData []testCase, b *testing.B) {
 	}
 }
 
+func benchmarkSetIfPresent(c Geche[string, string], testData []testCase, b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		if rand.Intn(2) == 0 {
+			c.Set(testData[i%len(testData)].key, "value")
```

Comment:
I don't quite understand this benchmark:
1. The result will be average of Set and SetIfPresent, which makes results hard to interpret.
2. You set odd keys, and set if present odd keys. Given key cardinality of 1_000_000 and total number of keys 10_000_000, the probability of `SetIfPresent` to hit "Present" case is 1/100. Which also makes it hard to interpret.

How about create separate benchmarks for case where key is present and a case where it does not?
In first case setting the keys should be before the main loop, and with `b.ResetTimer()` call, so data preparation is not included in the test.

Reply:
Ok let me do that. Do we want to include anything in the readme?

Reply:
You mean benchmark result? We'll need to rerun all benchmarks on a big machine.
No need for now. Too much hassle.
For the new interface method I think it makes sense to add its `SetIfNotPresent` counterpart first. Then we'll add to the docs and re-run benchmarks.